nibabel/filebasedimages.py

```diff
@@ -511,3 +512,79 @@ def path_maybe_image(klass, filename, sniff=None, sniff_max=1024):
         if sniff is None or len(sniff[0]) < klass._meta_sniff_len:
             return False, sniff
         return klass.header_class.may_contain_header(sniff[0]), sniff
+
+
+class SerializableImage(FileBasedImage):
+    '''
+    Abstract image class for (de)serializing images to/from byte strings.
+
+    The class doesn't define any image properties.
+
+    It has:
+
+    methods:
+
+       * .to_bytes() - serialize image to byte string
+
+    classmethods:
+
+       * from_bytes(bytestring) - make instance by deserializing a byte string
+
+    The following properties should hold:
+
+      * ``klass.from_bytes(bstr).to_bytes() == bstr``
```

Comment:
Is that true in general for round-tripping?  For example, what happens with nifti images with image scaling?  I think the scale factors may get recalculated.

Reply:
Fair point. I guess what we want is load-level idempotence, to the same level of precision as save/load cycles.:

```Python
img_a = klass.from_bytes(bstr)
img_b = klass.from_bytes(img_a.to_bytes())

np.allclose(img_a.get_fdata(), img_b.get_fdata())
```

Would the following be too strong, as well?

```Python
img_a.to_bytes() == img_b.to_bytes()
```